frontend/src/pages/registration/signUpPage.js

```diff
@@ -1,43 +1,112 @@
-import React, { useState } from "react";
-import { Link } from "react-router-dom";
+import React, {useContext, useEffect, useState} from "react";
+import {Link, useNavigate} from "react-router-dom";
+import axios from "axios";
+import apiEndpoints from "../../apiEndpoints";
+import {toast} from "sonner";
+import {AuthContext} from "./AuthContext";
+import AuthBtn from "../../components/containers/authBtn";
 
 function SignUpPage() {
-    function handleSubmit(e) {
+    const authContext = useContext(AuthContext);
+    const navigate = useNavigate();
+    const [userName, setUserName] = useState('');
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [repeatPassword, setRepPassword] = useState('');
+
+    const { login, isValidEmail, isValidUserName, isValidPassword } = authContext;
+
+    async function handleSubmit(e) {
         e.preventDefault();
 
-        const form = e.target;
-        const formData = new FormData(form);
+        if (!isValidUserName(userName)){
+            toast.error("Username can not contain such symbol or be empty!");
```

Comment:
yeah, we duplicate way too much. Same messages, function etc. Gotta do smth about it